packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -811,9 +821,16 @@ function commitExitViewTransitions(
         // Delete the entry so that we know when we've found all of them
         // and can stop searching (size reaches zero).
         appearingViewTransitions.delete(name);
+        // Note: If the other side ends up outside the viewport, we'll still run this.
+        // Therefore it's possible for onShare to be called with only an old snapshot.
+        scheduleViewTransitionEvent(deletion, props.onShare);
```

Comment:
How do we differentiate in `onShare` whether the instance is appearing or disappearing? PR description says `onShare` only runs on mount but here it runs on exit.